### PR TITLE
Handle non-digit characters in AuthorRetrieval().classificationgroup

### LINKED ISSUE #258: Classification group - invalid literal
```python3
from pybliometrics.scopus import AuthorRetrieval
scopus_id = "12792377500"
au = AuthorRetrieval(scopus_id, refresh=True)
print(au.classificationgroup)
```

```
$ python3 main.py 
Traceback (most recent call last):
  File "/mnt/dataint/data/progetti_miei/python/pybliometrics_bug/main.py", line 9, in <module>
    print(au.classificationgroup)
  File "/home/raffaele/.local/lib/python3.10/site-packages/pybliometrics/scopus/author_retrieval.py", line 66, in classificationgroup
    out = [(int(item['$']), int(item['@frequency'])) for item in
  File "/home/raffaele/.local/lib/python3.10/site-packages/pybliometrics/scopus/author_retrieval.py", line 66, in <listcomp>
    out = [(int(item['$']), int(item['@frequency'])) for item in
ValueError: invalid literal for int() with base 10: '2800;
```

### PR DESCRIPTION
**Pull request by raffaem**
fix #258

### CODE REVIEW

**Comment by Michael-E-Rose** [comment]
The `"".join(filter(str.isdigit, item["$"])` bit is good, but to wrap all of this in a try-except makes the code just too complex.

Why not just always use this expression? `filter()` is lightning fast and it can capture whatever non-digits there might be.

```
@property
def classificationgroup(self) -> Optional[List[Tuple[int, int]]]:
    """List with (subject group ID, number of documents)-tuples."""
    path = ['classificationgroup', 'classifications', 'classification']
    out = [(int("".join(filter(str.isdigit, item["$"])))
            int(item['@frequency']))
           for item in listify(chained_get(self._profile, path, []))]
    return out or None
```

**Comment by raffaem** [comment]
How does it look now?

**Comment by Michael-E-Rose** [comment]
Ah, you created an anonymous function for this. Also nice.

But why not making it a proper function? I have the feeling it might be used again later. That is, you just put the function in https://github.com/pybliometrics-dev/pybliometrics/blob/master/pybliometrics/scopus/utils/parse_content.py and import in this place. To comply with the names of the other function, what about filter_digits()?

File: pybliometrics/scopus/utils/parse_content.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 from functools import reduce
 from warnings import warn
 
+def filter_digits(s):
+    return "".join(filter(str.isdigit, s))
```
Comment (Michael-E-Rose) [inline pybliometrics/scopus/utils/parse_content.py:6]: Short docstring please

File: pybliometrics/scopus/author_retrieval.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from pybliometrics.scopus.superclasses import Retrieval
 from pybliometrics.scopus.utils import chained_get, check_parameter_value,\
     get_content, get_link, listify, make_int_if_possible, parse_affiliation,\
-    parse_date_created
+    parse_date_created, filter_digits
```
Comment (Michael-E-Rose) [inline pybliometrics/scopus/author_retrieval.py:12]: Sort alphabetically please

**Review by Michael-E-Rose — CHANGES_REQUESTED** [review]

**Review by Michael-E-Rose — APPROVED** [review]

**Comment by Michael-E-Rose** [comment]
Thanks for this!

**Pull request merged**